### PR TITLE
Update ratings endpoint to create cocktail if not already in DB

### LINKED ISSUE #7: Cocktail Rating
When a client sends a post "/api/v1/cocktails/:id/rating" to create a new rating, with a valid :id as an integer, the response status would be 201, 

```json
{
"data": { 
   "id": "id of rating",
   "type": "cocktail_rating",
   "attributes": {
      "user_id": "user id",
      "cocktail_id": "cocktail_id",
      "stars": "rating of the cocktail"
      }
   }
}
```

### PR DESCRIPTION
**Pull request by mkrumholz**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Rubocop has passed locally and any fixes were made for failures
- [ ] CircleCI build is passing
- [x] All tests passing locally

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: 
closes #7 

Rating endpoint just returned the cocktail with the new rating

## What is the new behavior?

- Now, if the cocktail being rated is not in our DB, we create the cocktail AND the rating, then return the cocktail with the new rating
- Standardized error messages
- Refactored some code

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/controllers/api/v1/cocktails/search_controller.rb [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+      search_results(params[:search], current_user.id)
+    end
+  end
 
-      if results == false
-        render json: { errors: "Couldn't find Cocktail" }, status: 404
-      else
-        render json: CocktailSearchSerializer.search_list(results)
-      end
+  def search_results(query, user_id)
```
Comment (markcyen) [inline app/controllers/api/v1/cocktails/search_controller.rb:10]: Sweet! Thanks for refactoring! 🧹

File: app/facades/cocktail_facade.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class CocktailFacade
   def self.retrieve_cocktail(user_id, cocktail_id)
-    return false if retrieve_details(cocktail_id) == {:drinks=>nil}
+    return false if retrieve_details(cocktail_id) == { drinks: nil }
```
Comment (markcyen) [inline app/facades/cocktail_facade.rb:3]: When I tried it this way, my tests were not passing so not sure what I was doing wrong...

File: app/models/rating.rb [hunk 3]
```diff
@@ -8,4 +8,10 @@ class Rating < ApplicationRecord
                       greater_than_or_equal_to: 0,
                       less_than_or_equal_to: 5
                     }
+
+  def self.prep_dataframe
+    Rating.all.select('user_id, cocktail_id, stars').map do |col|
```
Comment (markcyen) [inline app/models/rating.rb:13]: Sweet! 🍬

File: app/modules/dataframeable.rb [hunk 4]
```diff
@@ -1,18 +1,7 @@
...
   def self.create_df
-    all_three = Rating.all.select('user_id, cocktail_id, stars')
-
-    all_three_array = all_three.map do |three|
-      [three.user_id, three.cocktail_id, three.stars]
-    end
-
-    df = Pandas.DataFrame.new(data = all_three_array)
-
+    df = Pandas.DataFrame.new(data = Rating.prep_dataframe)
```
Comment (markcyen) [inline app/modules/dataframeable.rb:3]: So nice and simple...

**Review by markcyen — APPROVED** [review]
Looks good to me. 🍫👀

File: app/facades/cocktail_facade.rb (hunk 2 above)
Comment (mkrumholz) [reply app/facades/cocktail_facade.rb:3]: I think it is because I added/updated the retrieve_details method possibly?

Comment (mkrumholz) [reply app/facades/cocktail_facade.rb]: @markcyen yea, it's because now my retrieve_details method deals with nil values, so this can be more specific

**Review by markcyen — APPROVED** [review]
Good to go!

**Review by twarbelow — APPROVED** [review]
this looks good. circleci should be passing after merge.

**Pull request merged**